quickstart/troubleshooting.md

```diff
@@ -0,0 +1,353 @@
+Mar 09 10:44:57.866 info: MOAC is warmed up and ready to 🚀
+Mar 09 10:44:58.206 debug [csi]: probe request (ready=true)
+```
+{% endtab %}
+{% endtabs %}
+
+### mayastor's log file
+
+It is the data plane of Mayastor. There are as many mayastor containers as many
+storage nodes have been configured. This log file is useful when troubleshooting
+IO errors but not only that. Management operations could also fail because of
+a problem on the storage node.
+
+{% tabs %}
+{% tab title="Example obtaining mayastor's log" %}
+```sh
+kubectl -n mayastor logs mayastor-qgpw6 mayastor
+```
+{% endtab %}
+{% endtabs %}
+
+### mayastor CSI agent's log file
+
+When having a problem with (un)mounting volume on an application node, this log
+file can be useful. Generally all nodes in the cluster run mayastor CSI agent,
+so it's good to know which node is having the problem and inspect the log file
+only on that node.
+
+{% tabs %}
+{% tab title="Example obtaining mayastor CSI driver's log" %}
+```sh
+kubectl -n mayastor logs mayastor-csi-7pg82 mayastor-csi
+```
+{% endtab %}
+{% endtabs %}
+
+### CSI sidecars
+
+These containers implement CSI spec for Kubernetes and run in the same pods
+with moac and mayastor-csi containers. Although that they are not part of
+Mayastor software, can contain useful information when Mayastor CSI
+control/node plugin fails to register with k8s cluster.
+
+{% tabs %}
+{% tab title="Obtaining CSI control containers logs" %}
+```sh
+kubectl -n mayastor logs $(kubectl -n mayastor get pod -l app=moac -o jsonpath="{.items[0].metadata.name}") csi-attacher
+kubectl -n mayastor logs $(kubectl -n mayastor get pod -l app=moac -o jsonpath="{.items[0].metadata.name}") csi-provisioner
+```
+{% endtab %}
+{% endtabs %}
+
+{% tabs %}
+{% tab title="Example obtaining CSI node container log" %}
+```sh
+kubectl -n mayastor logs mayastor-csi-7pg82 csi-driver-registrar
+```
+{% endtab %}
+{% endtabs %}
+
+## Coredumps
+
+Coredump is a snapshot of process' memory with auxiliary information
+(PID, state of registers, etc.) saved to a file. It is used for post-mortem
+analysis and it is generated by the operating system in case of a severe error
+(i.e. memory corruption). Using a coredump for a problem analysis requires
+deep knowledge of program internals and is usually done only by developers.
+However, there is a very useful piece of information that even users can
+retrieve and this information alone can often identify the root cause of
+the problem. It is the stack (backtrace). Put differently, the thing that the
+program was doing at the time when it crashed. Here we describe how to get it.
+The steps apply to Ubuntu OS. However different Linux distros would use
+similar steps.
+
+We rely on systemd-coredump that saves and manages coredumps on the system,
+`coredumpctl` utility that is part of the same package and finally
+the `gdb` debugger.
+
+{% tabs %}
+{% tab title="Install systemd-coredump and gdb" %}
```

Comment:
what about lz4cat, is it already present on ubuntu?

Reply:
I will address this comment in following PR 👍